src/chsh.c

```diff
@@ -499,6 +501,10 @@ int main (int argc, char **argv)
 	 * name, or the name getlogin() returns.
 	 */
 	if (optind < argc) {
+		if (!is_valid_user_name (argv[optind])) {
+			fprintf (stderr, _("%s: Provided user name is not a valid name\n"), Prog);
+			fail_exit (E_NOPERM);
```

Comment:
I think we should exit with `1` or `EXIT_FAILURE`.

@hallyn  I see a lot of inconsistency in existing code.  Do you have any opinion?

Reply:
To keep some consistency I changed the exit code only in this one place. Other places are consistent with the files, where the conditions were added.